### PR TITLE
Docs: Adds "Debugging uncaught requests" recipe

### PR DESCRIPTION
**Pull request by kettanaito**
Provides a recipe for one of the most common issues.

### CODE REVIEW

File: docs/recipes/debugging-uncaught-requests.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+<PageLink
+  title="listen (setupServer)"
+  url="/docs/api/setup-server/listen#onunhandledrequest"
+/>
+
+## Examine the handlers
+
+- Applicable to: `setupWorker`/`setupServer`
+
+You can print out the list of current request handlers at any point of your application runtime or rest run. Browsing the list of handlers can help you figure out a request handler that doesn't get appended on runtime with `.use()`, or notice that the handler you thought is there is actually missing.
```
Comment (msutkowski) [inline docs/recipes/debugging-uncaught-requests.mdx:29]: What do you mean by 'or rest run'?

Comment (msutkowski) [inline docs/recipes/debugging-uncaught-requests.mdx:29]:
```suggestion
You can print out the list of current request handlers at any point of your application runtime or rest run. Browsing the list of handlers can help you determine if a request handler is not appended on runtime with `.use()`, or if the handler you thought is there is actually missing.
```

File: docs/recipes/debugging-uncaught-requests.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+worker.start({
+  serviceWorker: {
+    url: '/assets/mockServiceWorker.js',
+  },
+})
+```
+
+Then MSW will only intercept the requests under `/assets/*` path. A request to `/user` or `/posts` will not be intercepted as they lie **outside of the worker's scope** (`/assets`).
+
+> The Service Worker distributed by MSW is a regular Service Worker like any other. While not mandatory, the more you are familiar with how Service Workers function, the fewer issues you will experience. Learn more about [Using Service Worker (MDN)](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers).
```
Comment (msutkowski) [inline docs/recipes/debugging-uncaught-requests.mdx:59]:
```suggestion
> The Service Worker distributed by MSW is a regular Service Worker like any other. While not mandatory, the more familiar you are with how Service Workers function, the fewer issues you will experience. Learn more about [Using Service Worker (MDN)](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers).
```

File: docs/recipes/debugging-uncaught-requests.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  res.setHeader('Service-Worker-Allow', '/')
+  next()
+})
+```
+
+## Run in debug mode
+
+- Applicable to: `setupServer`
+
+Mock Service Worker supports [`debug`](https://github.com/visionmedia/debug) when run in NodeJS. You can leverage this to look into the internals of requests interception, as well as provide the output logs to the library maintainers for more details.
```
Comment (msutkowski) [inline docs/recipes/debugging-uncaught-requests.mdx:96]:
```suggestion
Mock Service Worker supports [`debug`](https://github.com/visionmedia/debug) when run in NodeJS. You can leverage this to look into the internals of how requests are intercepted, as well as provide the output logs to the library maintainers for more details.
```

File: docs/recipes/debugging-uncaught-requests.mdx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+Append the `DEBUG` environmental variable to the command that runs your tests:
+
+```bash
+$ DEBUG=* npm test
+```
+
+With that variable set, you will see a bunch of messages in your terminal indicating various stages of request interception that can hint you at what may have gone wrong.
```
Comment (msutkowski) [inline docs/recipes/debugging-uncaught-requests.mdx:104]:
```suggestion
With that variable set, you will see many messages in your terminal indicating various stages of request interception that can give you useful hints as to what may have gone wrong.
```

**Review by msutkowski — COMMENTED** [review]
I wonder if we should highlight [`findWorker`](https://mswjs.io/docs/api/setup-worker/start#findworker) anywhere in this. Overall, looks awesome!

File: docs/recipes/debugging-uncaught-requests.mdx (hunk 1 above)
Comment (kettanaito) [reply docs/recipes/debugging-uncaught-requests.mdx]: I've changed that to "during tests".

**Pull request merged**